src/jdk.compiler/share/classes/com/sun/tools/javac/parser/DocCommentParser.java

```diff
@@ -76,7 +76,16 @@ static class ParseException extends Exception {
         }
     }
 
-    private enum Phase { PREAMBLE, BODY, POSTAMBLE }
+    private enum Phase {
+        /** The initial part of an HTML file up to and including the {@code body} and possible {@code <main>} tag. */
+        PREAMBLE,
+        /** The initial part of a doc comment, or the rich-text content of a block tag. */
+        BODY,
+        /** The end of an HTML file, from and including the {@code </main>} or {@code </body>} tag. */
```

Comment:
Shouldn't there be the adjective _possible_ similarly to PREAMBLE? I assume, the idea here is as follows. The `body` element encloses the optional `main` element. If the `main` element is enclosed, then the PREAMBLE phase ends with the `main` start tag and the POSTAMBLE phase starts with the `main` end tag. Otherwise, PREAMBLE ends with the `body` start tag and POSTAMBLE starts with the `body` end tag. Is this correct?

Reply:
Yes, that's essentially correct, but the code is somewhat more tolerant.  
The wording was trying to express the implementation detail that PREAMBLE goes up to the `<body>` and then checks for `<main>`, whereas the BODY ends (and PREAMBLE begins) with _either_ `</main>` or `</body>`, whichever is found first.  In other words, we lazily accept a malformed file with `...<body><main>...</body>`.